src/main/java/mate/academy/spring/controller/UserController.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.spring.controller;
+
+import java.util.Optional;
+import mate.academy.spring.mapper.impl.response.UserResponseMapper;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private final UserService userService;
+    private final UserResponseMapper userResponseMapper;
+
+    public UserController(UserService userService,
+                          UserResponseMapper userResponseMapper) {
+        this.userService = userService;
+        this.userResponseMapper = userResponseMapper;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto findByEmail(@RequestParam String email) {
+        Optional<User> optionalUser = userService.findByEmail(email);
+        if (optionalUser.isPresent()) {
+            return userResponseMapper.toDto(optionalUser.get());
+        }
+        return new UserResponseDto();
```

Comment:
```suggestion
        if (optionalUser.isPresent()) {
            return userResponseMapper.toDto(optionalUser.get());
        }
        return new UserResponseDto();
```
Interesting. I think it's better to throw an Exception if the user is not found by the email.